app/assets/javascripts/pretty-text/addon/engines/discourse-markdown-it.js

```diff
@@ -159,11 +159,7 @@ function videoHTML(token) {
   const src = token.attrGet("src");
   const origSrc = token.attrGet("data-orig-src");
   const dataOrigSrcAttr = origSrc !== null ? `data-orig-src="${origSrc}"` : "";
-  return `<div class="video-container">
-    <video width="100%" height="100%" preload="metadata" controls>
-      <source src="${src}" ${dataOrigSrcAttr}>
-      <a href="${src}">${src}</a>
-    </video>
+  return `<div class="video-placeholder-container" data-video-src="${src}" ${dataOrigSrcAttr}>
```

Comment:
Later in this file, we have some code that outputs a different video placeholder for the preview: 

```
    if (
      options.discourse.previewing &&
      !options.discourse.limitedSiteSettings.enableDiffhtmlPreview
    ) {
      return `<div class="onebox-placeholder-container">
        <span class="placeholder-icon video"></span>
      </div>`;
    } else {
      return videoHTML(token);
    }
```

This is likely no longer needed... though the decorator needs to apply without `onlyStream: true` for it to work on the preview. Would be great if we could consolidate this and show the same element in both preview and post.

Reply:
Actually I think I just broke that for sites with enableDiffhtmlPreview enabled. It won't show the video in the composer preview anymore it will just show the new placeholder we just created.

Reply:
Okay, I made a change so that video playback in the composer will still work if `enableDiffhtmlPreview` is `true`. However I think we should keep the original placeholder for now because it isn't clickable and the new one looks like its clickable but clicking on it in the composer won't do anything.

Reply:
Ok! Would it be hard to have the a placeholder in the preview that is playable on click? I guess `api.decorateCookedElement` doesn't apply to the preview, so we'd need to replicate that logic elsewhere, right?

Reply:
That is the whole point of `enableDiffhtmlPreview`, so you can play videos in the composer and that still works just fine. We can't "just" make videos playable in the composer without whatever diffhtmlPreview is doing or every time you type things flicker (that's my current understanding anyways).

But ya the `api.decorateCookedElement` doesn't trigger in the composer, we'd have to replicate that logic somewhere else like where the image resizer logic lives.

Reply:
![CleanShot 2023-10-12 at 10 13 43](https://github.com/discourse/discourse/assets/1490496/6b86d575-3099-4716-8239-1ff408f686c0)

^ Video playback in the composer without `diffHtmlPreview`.